it/google-cloud-platform/src/test/java/org/apache/beam/it/gcp/pubsub/PubSubIOST.java

```diff
@@ -0,0 +1,510 @@
+      Primitive proto =
+          Primitive.newBuilder()
+              .setPrimitiveBytes(ByteString.copyFrom(byteValue))
+              .setPrimitiveInt32(ByteBuffer.wrap(byteValue).getInt())
+              .build();
+      context.output(proto);
+    }
+  }
+
+  /** Mapper class to convert data from KV<byte[], byte[]> to PubSubMessage. */
+  private static class MapKVtoPubSubMessage extends DoFn<KV<byte[], byte[]>, PubsubMessage> {
+    @ProcessElement
+    public void process(ProcessContext context) {
+      byte[] byteValue = Objects.requireNonNull(context.element()).getValue();
+      PubsubMessage pubsubMessage = new PubsubMessage(byteValue, Collections.emptyMap());
+      context.output(pubsubMessage);
+    }
+  }
+
+  /** Example of Generic class to test PubSubIO.writeAvros()/readAvros methods. */
+  static class GenericClass implements Serializable {
+    byte[] byteField;
+
+    public GenericClass() {}
+
+    public GenericClass(byte[] byteField) {
+      this.byteField = byteField;
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(getClass()).add("byteField", byteField).toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(Arrays.hashCode(byteField));
+    }
+
+    @Override
+    public boolean equals(@Nullable Object other) {
+      if (other == null || !(other instanceof GenericClass)) {
+        return false;
+      }
+      GenericClass o = (GenericClass) other;
+      return Arrays.equals(byteField, o.byteField);
+    }
+  }
+
+  private enum WriteAndReadFormat {
+    STRING,
+    AVRO,
+    PROTO,
+    PUBSUB_MESSAGE
+  }
+
+  /** Options for PubSub IO load test. */
+  static class Configuration extends SyntheticSourceOptions {
+    /** Pipeline timeout in minutes. Must be a positive value. */
+    @JsonProperty public int pipelineTimeout = 20;
+
+    /** Runner specified to run the pipeline. */
+    @JsonProperty public String runner = "DirectRunner";
+
+    /** PubSub write and read format: STRING/AVRO/PROTO/PUBSUB_MESSAGE. */
+    @JsonProperty public String writeAndReadFormat = "STRING";
+
+    /** Number of workers for the pipeline. */
+    @JsonProperty public int numWorkers = 20;
+
+    /** Maximum number of workers for the pipeline. */
+    @JsonProperty public int maxNumWorkers = 100;
+
+    /**
+     * Rate of generated elements sent to the source table. Will run with a minimum of 1k rows per
+     * second.
+     */
+    @JsonProperty public int rowsPerSecond = DEFAULT_ROWS_PER_SECOND;
+
+    /** Rows will be generated for this many minutes. */
+    @JsonProperty public int minutes = 15;
+
+    /**
+     * Determines the destination for exporting metrics. If set to true, metrics will be exported to
+     * InfluxDB and displayed using Grafana. If set to false, metrics will be exported to BigQuery
+     * and displayed with Looker Studio.
+     */
+    @JsonProperty public boolean exportMetricsToInfluxDB = true;
```

Comment:
If this is true but influxHost is not provided, could the test still run successfully (just skip export) ? From current logic it seems it will always call exportMetrics

Reply:
If some of the Influx parameters are not passed, then it automactically skips the exporting and test runs successfully. This logic is already implemented in `exportMetircs `method.